subgraphs/smolverse/src/mappings/smol-farm.ts

```diff
@@ -0,0 +1,114 @@
+import { store } from "@graphprotocol/graph-ts";
+import { SMOL_BRAINS_ADDRESS } from "@treasure/constants";
+import { log } from "matchstick-as";
+
+import { Claim, Random, StakedToken } from "../../generated/schema";
+import { RewardClaimed, SmolStaked, SmolUnstaked, StartClaiming } from "../../generated/Smol Farm/SmolFarm";
+import { SMOL_BRAINS_COLLECTION_NAME, SMOL_FARM_NAME, TOKEN_STANDARD_ERC721 } from "../helpers/constants";
+import { getFarmId, getRandomId, getStakedTokenId } from "../helpers/ids";
+import { getOrCreateCollection, getOrCreateFarm, getOrCreateToken, getOrCreateUser } from "../helpers/models";
+
+export function handleSmolStaked(event: SmolStaked): void {
+  const params = event.params;
+
+  const farm = getOrCreateFarm(event.address, SMOL_FARM_NAME);
+  const owner = getOrCreateUser(params._owner.toHexString());
+  const collection = getOrCreateCollection(SMOL_BRAINS_ADDRESS, SMOL_BRAINS_COLLECTION_NAME, TOKEN_STANDARD_ERC721);
+  const token = getOrCreateToken(collection, params._tokenId);
+
+  const stakedToken = new StakedToken(getStakedTokenId(token, farm));
+  stakedToken.token = token.id;
+  stakedToken.farm = farm.id;
+  stakedToken.owner = owner.id;
+  stakedToken.stakeTime = params._stakeTime;
+  stakedToken.claims = [];
+  stakedToken.save();
+}
+
+export function handleSmolUnstaked(event: SmolUnstaked): void {
+  const params = event.params;
+
+  const id = [
+    SMOL_BRAINS_ADDRESS.toHexString(),
+    params._tokenId.toHexString(),
+    getFarmId(event.address)
+  ].join("-");
+  const stakedToken = StakedToken.load(id);
+  if (!stakedToken) {
+    log.info("[smol-farm] Skipped already removed staked token: {}", [id]);
+    return;
+  }
+
+  for (let index = 0; index < stakedToken.claims.length; index++) {
+    store.remove("Claim", stakedToken.claims[index]);
+  }
+
+  store.remove("StakedToken", id);
+}
+
+export function handleStartClaiming(event: StartClaiming): void {
+  const params = event.params;
+
+  const stakedTokenId = [
+    SMOL_BRAINS_ADDRESS.toHexString(),
+    params._tokenId.toHexString(),
+    getFarmId(event.address)
+  ].join("-");
```

Comment:
Can you make a helper function for this? Looks like you do this multiple times.

Reply:
I've pushed a couple helpers for generating those IDs